app/src/main/java/com/taylorgirard/comicconvo/activities/IndividualMessageActivity.java

```diff
@@ -1,17 +1,213 @@
+                }
+
+                message.put("PairID", pairId);
+
+                message.saveInBackground(new SaveCallback() {
+                    @Override
+                    public void done(ParseException e) {
+                        if (e != null){
+                            Log.e(TAG, "Error saving message", e);
+                        } else{
+                            Log.d(TAG, "Successfully saved message");
+                        }
+                    }
+                });
+                etMessage.setText("");
+
+                user.addUnique("Messaged", match);
+                user.saveInBackground(new SaveCallback() {
+                    @Override
+                    public void done(ParseException e) {
+                        if (e != null){
+                            Log.e(TAG, "Error saving user messaged list", e);
+                        } else{
+                            Log.d(TAG, "Successfully saved user messaged list");
+                        }
+                    }
+                });
+
+                HashMap<String, Object> params = new HashMap<>();
+                params.put("user" , user.getObjectId());
+                params.put("match", match.getObjectId());
+                ParseCloud.callFunctionInBackground("saveMessaged", params, new FunctionCallback<Boolean>() {
+                    @Override
+                    public void done(Boolean object, ParseException e) {
+                        if (e==null && object){
+                            Log.i(TAG, "Successfully saved match messaged list");
+                        } else{
+                            Log.e(TAG, "Error saving match messaged list", e);
+                        }
+                    }
+                });
+            }
+        });
     }
+
+    void loadMessages(){
+
+        String websocketUrl = "wss://comicconvo.b4a.io/";
+
+        ParseLiveQueryClient parseLiveQueryClient = null;
+        try {
+            parseLiveQueryClient = ParseLiveQueryClient.Factory.getClient(new URI(websocketUrl));
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        }
+
+        List<ParseQuery<Message>> queryList = new ArrayList<>();
+
+        ParseQuery<Message> querySender= ParseQuery.getQuery(Message.class);
+        querySender.whereEqualTo("Sender", user);
+        querySender.whereEqualTo("Receiver", match);
+        queryList.add(querySender);
+
+        ParseQuery<Message> queryReceiver= ParseQuery.getQuery(Message.class);
+        queryReceiver.whereEqualTo("Sender", match);
+        queryReceiver.whereEqualTo("Receiver", user);
+        queryList.add(queryReceiver);
+
+        ParseQuery<Message> finalQuery = ParseQuery.or(queryList);
+
+        SubscriptionHandling<Message> subscriptionHandling = parseLiveQueryClient.subscribe(finalQuery);
+
+        subscriptionHandling.handleEvent(SubscriptionHandling.Event.CREATE, (query, object) -> {
+            messages.add(0, object);
```

Comment:
do we handle moving existing messages being updated?

Reply:
I'm not quite sure what you mean, but messages that have already been sent cannot be edited